Basic/Core/Core.pm

```diff
@@ -1142,7 +1142,9 @@ sub _establish_type {
   return $PDL_CD if UNIVERSAL::isa($item, 'Math::Complex');
   return max($item->type->enum, $sofar) if UNIVERSAL::isa($item, 'PDL');
   return $PDL_D if ref($item) ne 'ARRAY';
-  max($sofar, map _establish_type($_, $sofar), @$item);
+  #  only need to recurse for items that are refs
+  #  as $sofar will be $PDL_D at a minimum
+  max ($sofar, map {_establish_type($_, $sofar)} grep {ref} @$item);
```

Comment:
Please don't change the `map thing($_), @list` syntax back to `map {thing($_)} @list`, it adds an unnecessary scope-creating Perl op which slows things down. The same syntax is available for `grep` too.

Reply:
Done in b57df43c2c2685f9aeb07ce98e9f0219b7417e7f

Reply:
Thanks!